### PR TITLE
[framework] Entity factories now all uses EntityNameResolver

### LINKED ISSUE #622: All entity factories should use EntityNameResolver
Resolving entity in factories is great, but not all Factories are doing that, for example `GoogleFeedItemFactory`.

### Expected result
All factories are using EntityNameResolver instead of creating specific Entity.

### PR DESCRIPTION
**Pull request by MattCzerner**
| Q             | A
| ------------- | ---
|Description, reason for the PR| All entity factories should use EntityNameResolver as was decided here #622 
|New feature| No
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md) | Yes
|Fixes issues| close #622 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by boris-brtan** [comment]
this is BC-break

File: packages/product-feed-google/src/Model/FeedItem/GoogleFeedItemFactory.php [hunk 1]
```diff
@@ -12,6 +13,11 @@
 
 class GoogleFeedItemFactory
```
[the remark this answers is not in the document]
Comment (boris-brtan) [reply packages/product-feed-google/src/Model/FeedItem/GoogleFeedItemFactory.php]:
```php

        $classes = array();
        $metas = $em->getMetadataFactory()->getAllMetadata();
        foreach ($metas as $meta) {
            $classes[] = $meta->getName();
        }
        d($classes);die;
```
but that is strange that not all entities got factories right now...
```
Shopsys\FrameworkBundle\Model\Article\Article"
  1 => "Shopsys\FrameworkBundle\Model\Administrator\Administrator"
  2 => "Shopsys\FrameworkBundle\Model\Category\Category"
  3 => "Shopsys\FrameworkBundle\Model\Customer\User"
  4 => "Shopsys\FrameworkBundle\Model\Order\Order"
  5 => "Shopsys\FrameworkBundle\Model\Order\Item\OrderItem"
  6 => "Shopsys\FrameworkBundle\Model\Transport\Transport"
  7 => "Shopsys\FrameworkBundle\Model\Payment\Payment"
  8 => "Shopsys\FrameworkBundle\Model\Product\Product"
  9 => "Shopsys\FrameworkBundle\Model\Product\Brand\Brand"
  10 => "Shopsys\FrameworkBundle\Model\Pricing\Currency\Currency"
  11 => "Shopsys\FrameworkBundle\Model\Pricing\Group\PricingGroup"
  12 => "Shopsys\FrameworkBundle\Model\Pricing\Vat\Vat"
  13 => "Shopsys\FrameworkBundle\Model\Category\TopCategory\TopCategory"
  14 => "Shopsys\FrameworkBundle\Model\Category\CategoryTranslation"
  15 => "Shopsys\FrameworkBundle\Model\Category\CategoryDomain"
  16 => "Shopsys\FrameworkBundle\Model\Slider\SliderItem"
  17 => "Shopsys\FrameworkBundle\Model\Mail\MailTemplate"
  18 => "Shopsys\FrameworkBundle\Model\Transport\TransportPrice"
  19 => "Shopsys\FrameworkBundle\Model\Transport\TransportDomain"
  20 => "Shopsys\FrameworkBundle\Model\Transport\TransportTranslation"
  21 => "Shopsys\FrameworkBundle\Model\Payment\PaymentDomain"
  22 => "Shopsys\FrameworkBundle\Model\Payment\PaymentPrice"
  23 => "Shopsys\FrameworkBundle\Model\Payment\PaymentTranslation"
  24 => "Shopsys\FrameworkBundle\Model\Module\EnabledModule"
  25 => "Shopsys\FrameworkBundle\Model\Script\Script"
  26 => "Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCode"
  27 => "Shopsys\FrameworkBundle\Model\Order\OrderNumberSequence"
  28 => "Shopsys\FrameworkBundle\Model\Order\Status\OrderStatusTranslation"
  29 => "Shopsys\FrameworkBundle\Model\Order\Status\OrderStatus"
  30 => "Shopsys\FrameworkBundle\Model\Advert\Advert"
  31 => "Shopsys\FrameworkBundle\Model\Product\Pricing\ProductManualInputPrice"
  32 => "Shopsys\FrameworkBundle\Model\Product\Pricing\ProductCalculatedPrice"
  33 => "Shopsys\FrameworkBundle\Model\Product\Accessory\ProductAccessory"
  34 => "Shopsys\FrameworkBundle\Model\Product\Availability\Availability"
  35 => "Shopsys\FrameworkBundle\Model\Product\Availability\AvailabilityTranslation"
  36 => "Shopsys\FrameworkBundle\Model\Product\Brand\BrandDomain"
  37 => "Shopsys\FrameworkBundle\Model\Product\Brand\BrandTranslation"
  38 => "Shopsys\FrameworkBundle\Model\Product\TopProduct\TopProduct"
  39 => "Shopsys\FrameworkBundle\Model\Product\Flag\FlagTranslation"
  40 => "Shopsys\FrameworkBundle\Model\Product\Flag\Flag"
  41 => "Shopsys\FrameworkBundle\Model\Product\Parameter\Parameter"
  42 => "Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterTranslation"
  43 => "Shopsys\FrameworkBundle\Model\Product\Parameter\ParameterValue"
  44 => "Shopsys\FrameworkBundle\Model\Product\Parameter\ProductParameterValue"
  45 => "Shopsys\FrameworkBundle\Model\Product\BestsellingProduct\ManualBestsellingProduct"
  46 => "Shopsys\FrameworkBundle\Model\Product\Unit\UnitTranslation"
  47 => "Shopsys\FrameworkBundle\Model\Product\Unit\Unit"
  48 => "Shopsys\FrameworkBundle\Model\Product\ProductDomain"
  49 => "Shopsys\FrameworkBundle\Model\Product\ProductVisibility"
  50 => "Shopsys\FrameworkBundle\Model\Product\ProductCategoryDomain"
  51 => "Shopsys\FrameworkBundle\Model\Product\ProductTranslation"
  52 => "Shopsys\FrameworkBundle\Model\Cart\Item\CartItem"
  53 => "Shopsys\FrameworkBundle\Model\Cart\Cart"
  54 => "Shopsys\FrameworkBundle\Model\Administrator\Activity\AdministratorActivity"
  55 => "Shopsys\FrameworkBundle\Model\Administrator\AdministratorGridLimit"
  56 => "Shopsys\FrameworkBundle\Model\Country\CountryDomain"
  57 => "Shopsys\FrameworkBundle\Model\Country\Country"
  58 => "Shopsys\FrameworkBundle\Model\Country\CountryTranslation"
  59 => "Shopsys\FrameworkBundle\Model\Customer\BillingAddress"
  60 => "Shopsys\FrameworkBundle\Model\Customer\DeliveryAddress"
  61 => "Shopsys\FrameworkBundle\Model\PersonalData\PersonalDataAccessRequest"
  62 => "Shopsys\FrameworkBundle\Model\Newsletter\NewsletterSubscriber"
  63 => "Shopsys\FrameworkBundle\Component\Cron\CronModule"
  64 => "Shopsys\FrameworkBundle\Component\Image\Image"
  65 => "Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrl"
  66 => "Shopsys\FrameworkBundle\Component\Setting\SettingValue"
  67 => "Shopsys\FrameworkBundle\Component\DataFixture\PersistentReference"
  68 => "Shopsys\FrameworkBundle\Component\UploadedFile\UploadedFile"
  69 => "Shopsys\ShopBundle\Model\Article\Article"
  70 => "Shopsys\ShopBundle\Model\Administrator\Administrator"
  71 => "Shopsys\ShopBundle\Model\Category\Category"
  72 => "Shopsys\ShopBundle\Model\Customer\User"
  73 => "Shopsys\ShopBundle\Model\Order\Order"
  74 => "Shopsys\ShopBundle\Model\Order\Item\OrderItem"
  75 => "Shopsys\ShopBundle\Model\Transport\Transport"
  76 => "Shopsys\ShopBundle\Model\Payment\Payment"
  77 => "Shopsys\ShopBundle\Model\Product\Product"
  78 => "Shopsys\ShopBundle\Model\Product\Brand\Brand"
  79 => "Shopsys\ProductFeed\HeurekaBundle\Model\Product\HeurekaProductDomain"
  80 => "Shopsys\ProductFeed\HeurekaBundle\Model\HeurekaCategory\HeurekaCategory"
  81 => "Shopsys\ProductFeed\ZboziBundle\Model\Product\ZboziProductDomain"
  82 => "Shopsys\ProductFeed\GoogleBundle\Model\Product\GoogleProductDomain
```

[the remark this answers is not in the document]
Comment (MattCzerner) [reply packages/product-feed-google/src/Model/FeedItem/GoogleFeedItemFactory.php]: So you suppose not to use entity name resolver in all Feed classes right?

File: packages/framework/src/Component/Image/ImageFactory.php [hunk 2]
```diff
@@ -19,11 +25,16 @@ class ImageFactory implements ImageFactoryInterface
...
 
     /**
+     * @param \Shopsys\FrameworkBundle\Component\EntityExtension\EntityNameResolver $entityNameResolver
      * @param \Shopsys\FrameworkBundle\Component\Image\Processing\ImageProcessor $imageProcessor
      * @param \Shopsys\FrameworkBundle\Component\FileUpload\FileUpload $fileUpload
      */
-    public function __construct(ImageProcessor $imageProcessor, FileUpload $fileUpload)
-    {
+    public function __construct(
+        EntityNameResolver $entityNameResolver,
```
[the remark this answers is not in the document]
Comment (MattCzerner) [reply packages/framework/src/Component/Image/ImageFactory.php]: just a alphabetical sort. But it is true that other parts of the application has it as its last. So I will too, I always wonder where to add a dependencies. I would prefer a alphabetical sort of some kind or something else.

File: packages/product-feed-google/src/Model/FeedItem/GoogleFeedItemFactory.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (MattCzerner) [reply packages/product-feed-google/src/Model/FeedItem/GoogleFeedItemFactory.php]: Also, HeurekaProductDomain or ZboziProductDomain should use the factories too.

[the remark this answers is not in the document]
Comment (boris-brtan) [reply packages/product-feed-google/src/Model/FeedItem/GoogleFeedItemFactory.php]: I think that the thing that don't have factories are ok.

**Pull request merged**